lib/mysqlService.js

```diff
@@ -231,6 +221,59 @@ class MysqlService {
         }
     }
 
+
+    /**
+     * Stream the results of a query.
+     * @param {String} sql
+     * @param {ReadableOptions} [streamOptions] Options to be passed to the stream constructor.
+     * @return {stream.Readable}
+     */
+    streamQuery(sql, streamOptions = {}) {
+      return this._streamQuery(this._getReadPool(), sql, streamOptions);
+    }
+
+    /**
+     * Stream the results of a query from the master pool
+     * @param {String} sql
+     * @param {ReadableOptions} [streamOptions] Options to be passed to the stream constructor.
+     * @return {stream.Readable}
+     */
+    streamQueryFromMaster(sql, streamOptions = {}) {
+      return this._streamQuery(this._writePool, sql, streamOptions);
+    }
+
+
+    /**
+     * @param pool
+     * @param {String} sql
+     * @param  {ReadableOptions} streamOptions
+     * @return {stream.Readable}
+     */
+    _streamQuery(pool, sql, streamOptions) {
+      this._logSqlQuery(sql);
+
+      const defaults = { objectMode: true };
+      const opts = Object.assign(defaults, streamOptions);
+
+      /** @type {stream.Readable} **/
+      const queryStream = pool.query(sql)
+        .on('error', err => this._logSqlError(err, sql))
+        .stream(opts);
+
+      const transform = new MysqlService._TransformResult(opts);
+      return queryStream.pipe(transform);
+    }
+
+    static get _TransformResult() {
+      class TransformResult extends TransformStream {
+        _transform(data, encoding, cb) {
+          // convert RowDataPacket to a standard JS object
+          cb(null, Object.assign({}, data));
```

Comment:
Looks like you use Object.assign({}, data) twice with that comment. Maybe just roll it into a private function called `_convertToJsObject(row)` which will DRY up and use the function to comment.